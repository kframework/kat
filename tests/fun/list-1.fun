// 3

// testing letrec and multiple arguments with currying

letrec head [ h : t ] = h
   and tail [ h : t ] = [ t ]
   and second l x     = head (tail l)
in second [ 1 : 3 : 5 : 2 : 4 : 0 : -1 : -5 : .Exps ] true
