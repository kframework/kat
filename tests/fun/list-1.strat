79debae7059d7c14cd35ad21a12436ea (^ regular | #normal | #branch) *
# xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx2 (^ regular | #normal | #branch) * ; abstract
# xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx3 (^ regular | #normal | #branch) * ; abstract ; #loop
# xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx4 (^ regular | #normal | #branch) * ; abstract ; #loop ; (^ regular | #normal | #branch) *
0a8a4195b744b13d578182b0a9a4981c orig
c2ea9004326fea4b31bb1cfbc2552ff1 exec
# xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx7 compile
