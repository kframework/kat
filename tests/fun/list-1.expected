
Strategy: (^ regular | #normal | #branch) *
================================================================================
<kat-FUN>
  <s>
    #STUCK ( )
  </s>
  <kat>
    <analysis>
      .Analysis
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <FUN>
      <k>
        mu ( .Names , head ( tail l ) ) ~> popEnv ~> popEnv ~> popEnv
      </k>
      <env>
        head |-> closure ( .Map , [ h : t ] -> mu ( head , tail , second , .Names , h ) | .Cases )
        l |-> [ 1 : 3 : 5 : 2 : 4 : 0 : -1 : -5 : .Vals ]
        second |-> closure ( .Map , l x -> mu ( head , tail , second , .Names , head ( tail l ) ) | .Cases )
        tail |-> closure ( .Map , [ h : t ] -> mu ( head , tail , second , .Names , [ t ] ) | .Cases )
        x |-> true
      </env>
      <envs>
        ListItem ( .Map )
        ListItem ( head |-> closure ( .Map , [ h : t ] -> mu ( head , tail , second , .Names , h ) | .Cases )
        second |-> closure ( .Map , l x -> mu ( head , tail , second , .Names , head ( tail l ) ) | .Cases )
        tail |-> closure ( .Map , [ h : t ] -> mu ( head , tail , second , .Names , [ t ] ) | .Cases ) )
        ListItem ( .Map )
      </envs>
    </FUN>
  </harness>
</kat-FUN>

Strategy: # (^ regular | #normal | #branch) * ; abstract
================================================================================
SKIPPING

Strategy: # (^ regular | #normal | #branch) * ; abstract ; #loop
================================================================================
SKIPPING

Strategy: # (^ regular | #normal | #branch) * ; abstract ; #loop ; (^ regular | #normal | #branch) *
================================================================================
SKIPPING

Strategy: orig
================================================================================
<FUN>
  <k>
    3
  </k>
  <env>
    .Map
  </env>
  <envs>
    .List
  </envs>
</FUN>

Strategy: exec
================================================================================
<kat-FUN>
  <s>
    #STUCK ( )
  </s>
  <kat>
    <analysis>
      .Analysis
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <FUN>
      <k>
        3
      </k>
      <env>
        .Map
      </env>
      <envs>
        .List
      </envs>
    </FUN>
  </harness>
</kat-FUN>

Strategy: # compile
================================================================================
SKIPPING
